### PR TITLE
Enable systemd service to create fileshare symlink

### PR DESCRIPTION
**Pull request by jsf9k**
## 🗣 Description ##

This pull request makes the change of enabling a `systemd` service to create the symlink to the AWS EFS file share.  It also creates some PCA-specific environment variables that were asked for in cisagov/cool-system#212.

## 💭 Motivation and context ##

Previously this was done via an `xdg-autostart` job, but as described in cisagov/ansible-role-xfce-cool#17 this does not create a _directly_ usable symlink.  It is not possible to drag and drop files on top of the symlink in the graphical desktop.

The changes in this pull request require the changes made in cisagov/ansible-role-xfce-cool#20, and is therefore the sister PR to that one and hence resolves cisagov/ansible-role-xfce-cool#17.

Resolves cisagov/cool-system#212.

See also cisagov/teamserver-packer#36.

## 🧪 Testing ##

I build a new AMI for our COOL staging environment with these changes and verified that it functions as expected.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - _eschew scope creep!_
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated to reflect the changes in this PR.
- [x] All new and existing tests pass.
- [x] Build and test a new staging AMI with these changes.

## ✅ Pre-merge checklist ##

- [x] Revert to default branch of `xfce_cool` Ansible role after cisagov/ansible-role-xfce-cool#20 is approved and merged.
- [x] Finalize version.
- [x] Build new production AMI with these changes.

### CODE REVIEW

**Review by dav3r — APPROVED** [review]
Nice improvement! 👍 

We don't yet have a way to automatically make the EFS share (`/share`) owned by the VNC user, right?  I believe that still has to be done manually by an operator (https://github.com/cisagov/cool-system/issues/109).

**Comment by jsf9k** [comment]
> We don't yet have a way to automatically make the EFS share (`/share`) owned by the VNC user, right? I believe that still has to be done manually by an operator ([cisagov/cool-system#109](https://github.com/cisagov/cool-system/issues/109)).

I could do that as a system-mode `systemd` service on each instance.  The problem is that they might clobber each other since the `vnc` user need not have the same user ID on every machine.

**Comment by dav3r** [comment]
> > We don't yet have a way to automatically make the EFS share (`/share`) owned by the VNC user, right? I believe that still has to be done manually by an operator ([cisagov/cool-system#109](https://github.com/cisagov/cool-system/issues/109)).
> 
> I could do that as a system-mode `systemd` service on each instance. The problem is that they might clobber each other since the `vnc` user need not have the same user ID on every machine.

Is there a way to do this based on the user name (which is known when we build the AMI) and not the UID?

**Comment by jsf9k** [comment]
> > > We don't yet have a way to automatically make the EFS share (`/share`) owned by the VNC user, right? I believe that still has to be done manually by an operator ([cisagov/cool-system#109](https://github.com/cisagov/cool-system/issues/109)).
> > 
> > 
> > I could do that as a system-mode `systemd` service on each instance. The problem is that they might clobber each other since the `vnc` user need not have the same user ID on every machine.
> 
> Is there a way to do this based on the user name (which is known when we build the AMI) and not the UID?

I don't think so.  In Linux the user and group names are just a sugarcoating to make things easier for humans. Under the hood it's UIDs and GIDs.  I think [this](https://docs.aws.amazon.com/efs/latest/ug/accessing-fs-nfs-permissions.html#accessing-fs-nfs-permissions-access-points) is the real solution for this problem.

**Comment by dav3r** [comment]
> > > > We don't yet have a way to automatically make the EFS share (`/share`) owned by the VNC user, right? I believe that still has to be done manually by an operator ([cisagov/cool-system#109](https://github.com/cisagov/cool-system/issues/109)).
> > > 
> > > 
> > > I could do that as a system-mode `systemd` service on each instance. The problem is that they might clobber each other since the `vnc` user need not have the same user ID on every machine.
> > 
> > 
> > Is there a way to do this based on the user name (which is known when we build the AMI) and not the UID?
> 
> I don't think so. In Linux the user and group names are just a sugarcoating to make things easier for humans. Under the hood it's UIDs and GIDs. I think [this](https://docs.aws.amazon.com/efs/latest/ug/accessing-fs-nfs-permissions.html#accessing-fs-nfs-permissions-access-points) is the real solution for this problem.

This could also be done with a simple cloud-init shell script that runs after the EFS share has been mounted, no?

**Comment by jsf9k** [comment]
> > > > > We don't yet have a way to automatically make the EFS share (`/share`) owned by the VNC user, right? I believe that still has to be done manually by an operator ([cisagov/cool-system#109](https://github.com/cisagov/cool-system/issues/109)).
> > > > 
> > > > 
> > > > I could do that as a system-mode `systemd` service on each instance. The problem is that they might clobber each other since the `vnc` user need not have the same user ID on every machine.
> > > 
> > > 
> > > Is there a way to do this based on the user name (which is known when we build the AMI) and not the UID?
> > 
> > 
> > I don't think so. In Linux the user and group names are just a sugarcoating to make things easier for humans. Under the hood it's UIDs and GIDs. I think [this](https://docs.aws.amazon.com/efs/latest/ug/accessing-fs-nfs-permissions.html#accessing-fs-nfs-permissions-access-points) is the real solution for this problem.
> 
> This could also be done with a simple cloud-init shell script that runs after the EFS share has been mounted, no?

Yes, but again the different machines could clobber each other.  The best solution is:

1. Fix the `vnc` UID and GID when we create the AMIs.  The main thing here is choosing something that is from the intersection of the user account ranges used by Debian, Kali, and Fedora.
2. Create an EFS access point or use `systemd` or use `cloud-init`.

File: src/vnc.yml [hunk 1]
```diff
@@ -1,31 +1,101 @@
 ---
 - hosts: all
-  name: Install VNC
+  name: Install VNC and configure vnc user
```
Comment (mcdonnnj) [inline src/vnc.yml:3]: `vnc user` and `VNC user` are both being used. I would like us to pick one and use it consistently.

**Review by mcdonnnj — APPROVED** [review]
Thanks again for figuring this out. Another request for consistency but otherwise LGTM ✔

File: src/vnc.yml (hunk 1 above)
Comment (dav3r) [reply src/vnc.yml:3]: I prefer `VNC user`, since it the username is configurable.

Comment (jsf9k) [reply src/vnc.yml]: See commit 06e0f17.

**Pull request merged**